src/api/spec/lib/feature_switch/feature_spec.rb

```diff
@@ -1,25 +1,35 @@
 require 'rails_helper'
 
 RSpec.describe Feature do
-  let(:user) { create(:confirmed_user, :in_beta, login: 'Tom') }
-  subject { Feature.active?('bootstrap') }
+  subject { Feature.active?(:bootstrap) }
+
+  let(:is_beta) { false }
```

Comment:
isn't that just a different way of Mock?

Reply:
it's a variable depending on context

Reply:
which turns out to be exactly the same as stub the method ```is_beta?``` from the user, no?

Reply:
you confuse me.

This PR decouples the whole thing from the user model - it's adding an explicit switch to be called from controllers. It's basically a policy on what features to use - the features don't peek into a global variable anymore, but take what was told at the beginning.

Reply:
Actually you confuse me, I'm trying to understand your reasoning. In the PR https://github.com/openSUSE/open-build-service/pull/7751 I understood that you want always to test the models interacting with the SUT, and that tests should avoid regression. So if somebody delete the method `User#is_beta?` this test should be able to catch it. Will this spec be able to catch it and avoid this regression? Or using your [Bible](https://martinfowler.com/articles/mocksArentStubs.html#TestIsolation)  as reference: _"If you introduce a bug to a system with mockist testing, it will usually cause only tests whose SUT contains the bug to fail._"

Reply:
but the Feature library has no relation to User model - that's what the whole thing is fixing.

If User#is_beta? exists or not does not matter to the feature switch - as long as someone calls use_beta_features with true at one point. It might be a global instance setting in the future - the feature library won't care.

Stubbing means hiding the layer *below* the code under test - but User is not there. So no, I don't need to create artificial dependencies in tests to avoid stubbing.